### PR TITLE
feat(studies): SKFP-874 export authorized studies widget to ferlab

### PR DESCRIPTION
**Pull request by lflangis**
# FEAT

- closes #[874](https://d3b.atlassian.net/browse/SKFP-874)

## Description
As a user with an eRA Commons login, I want to view and manage my access to controlled access studies in the Kids First (KF) data portal. Associated Registered/open access files will also be included for the controlled access studies that have both. The goal is to simplify the code complexity associated with retrieving data displayed on the front-end by pulling directly from the backend. In hopes to facilitate the handling of this issue: SKFP-868: [Authorized Studies] NCI Fence Issues
BACKLOG
 

## Acceptance Criterias
**Widget Title and Information:**

The widget title should display the number of controlled access studies accessible by the user.
image

An information icon popover should provide details about accessing controlled data with an eRA Commons account.
image

**Messaging Above List:**
Display an information icon with a popover explaining the user's access to controlled data.

Include a link to "Manage your connections," opening a modal for fence selection.
image

**Fence Selection Modal:**

The modal should allow users to manage connections to Kids First Framework Services (repository = gen3) and NCI CRDC Framework Services (repository = dcf).
image

Provide information about connecting accounts and following data use limitations.
image

**Fence Selection State:**

Display the state based on fence connections (connected/disconnected). Indicate which fence(s) are connected and display controlled access studies permitted by the user for each fence.
image

**Empty Widget States:**

Display a message if both fences are disconnected, guiding the user to connect using NIH credentials.
image

If at least one fence is connected but no accesses, inform the user and provide a link to apply for data access.
image

**Widget State with Study List:**

List controlled access studies based on the connected fence(s).
Display study names with study codes, providing a tooltip for long names.
Include a line describing the number of files the user has access to based on consent code/acl and number of registered access files when it applies to the study.
Provide links to redirect to Data Exploration for further details on access.
List consent codes/acls associated with the user's permissions for each study.
Include a progress bar highlighting the completeness of user access.
image

## Screenshot
image

https://github.com/Ferlab-Ste-Justine/ferlab-ui/assets/65532894/c51e196d-056b-4d51-8004-48f88e4e4119

### CODE REVIEW

File: packages/ui/Release.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+### 7.18.0 2023-12-14
+- feat: CLIN-1934 add custom pill
```
Comment (AltefrohneGaelle) [inline packages/ui/Release.md:2]: tu as juste copié collé 😄 il faudrait modifier la date et le ticket / description

File: packages/ui/src/components/AuthorizedStudies/AuthorizedStudiesListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
+//import { numberWithCommas } from 'utils/string';
```
Comment (AltefrohneGaelle) [inline packages/ui/src/components/AuthorizedStudies/AuthorizedStudiesListItem.tsx:1]: clean

File: packages/ui/src/components/PopoverContentLink/index.module.scss [hunk 3]
```diff
@@ -0,0 +1,9 @@
+@import 'theme.template';
+
+.popoverContentLink {
+  padding: 0;
+  &,
```
Comment (AltefrohneGaelle) [inline packages/ui/src/components/PopoverContentLink/index.module.scss:5]: ça veut dire quoi le &, ?

File: packages/ui/Release.md (hunk 1 above)
Comment (lflangis) [reply packages/ui/Release.md:2]: ahah plus un rebase interactif ou j'ai oublié de mettre le commentaire que j'avais fait

File: packages/ui/src/components/PopoverContentLink/index.module.scss (hunk 3 above)
Comment (lflangis) [reply packages/ui/src/components/PopoverContentLink/index.module.scss:5]: J'ai repris ce qui était fait dans k-f, j'ai retiré je pense que ca ne sert a rien dans ce context

**Review by AltefrohneGaelle — APPROVED** [review]

**Pull request merged**